### PR TITLE
fix(@desktop/wallet): adjust collectible details screen to design

### PR DESCRIPTION
**Pull request by dlipicar**
#5076

### What does the PR do

This PR completes [#5076](https://github.com/status-im/status-desktop/issues/5076)

Reordered several items 

Depends on https://github.com/status-im/status-go/pull/2984

### Affected areas

Wallet

### StatusQ checklist

Tested several different cases, resized window to check everything works as expected (particularly of interest: large number of properties and/or long description)
Tested on light and dark theme

### Screenshot of functionality (including design for comparison)

Design: https://www.figma.com/file/FkFClTCYKf83RJWoifWgoX/Wallet-v2?node-id=6455%3A123667&t=aBz9k21RQUEWxgEt-0

image

image

image

PR:

image

image

image

[collectible_details-2022-11-28_18.13.24.webm](https://user-images.githubusercontent.com/11161531/204382386-38cdb53c-8404-4ea1-b6da-51ef35701d88.webm)

I've checked the design and this PR matches it, except for:
- The design shows network name "Ethereum", while the backend returns name "Ethereum Mainnet" (Should I handle this?)

### CODE REVIEW

**Comment by anastasiyaig** [comment]
@dlipicar please move the PR to ready for review (in case it is ready, i assumed so because u added myself). Also, please bypass dev reviews and address them , so i can test the PR 
thank you!

File: ui/app/AppLayouts/Wallet/views/RightTabView.qml [hunk 1]
```diff
@@ -29,7 +29,7 @@ Item {
         function getBackButtonText(index) {
             switch(index) {
                 case 1:
-                return qsTr("Assets")
+                return qsTr("Collectibles")
```
Comment (Khushboo-dev-cpp) [inline ui/app/AppLayouts/Wallet/views/RightTabView.qml:32]: design says assets , I do think thats wrong but can we double check with someone from Design?

https://www.figma.com/file/FkFClTCYKf83RJWoifWgoX/Wallet-v2?node-id=6455%3A126137&t=5AeHPRDxszbKxt7N-4

File: ui/app/AppLayouts/Wallet/controls/CollectibleDetailsHeader.qml [hunk 2]
```diff
@@ -8,41 +8,87 @@ import StatusQ.Core.Theme 0.1
...
+    property string secondaryText
+    property bool isNarrowMode
+    property string networkName
+    property string networkColor
+    property string networkIconURL
+
     property StatusAssetSettings asset: StatusAssetSettings {
-        width: 40
-        height: 40
+        readonly property int size: root.isNarrowMode ? 24 : 38
```
Comment (Khushboo-dev-cpp) [inline ui/app/AppLayouts/Wallet/controls/CollectibleDetailsHeader.qml:23]: we have had multiple discussion on how to handle smaller screen sizes and I think we were to check one-one basis. however making assets size smaller and bigger based on screen size doesnt seem ideal. but this needs to come from design, so again, we double check?

File: ui/app/AppLayouts/Wallet/controls/CollectibleDetailsHeader.qml [hunk 3]
```diff
@@ -8,41 +8,87 @@ import StatusQ.Core.Theme 0.1
...
+        width: size
+        height: size
         isImage: true
     }
 
+    Component {
+        id: collectibleIdComponent
+        StatusBaseText {
+            Layout.alignment: Qt.AlignVCenter | Qt.AlignLeft
+            font.pixelSize: isNarrowMode ? 15 : 22
```
Comment (Khushboo-dev-cpp) [inline ui/app/AppLayouts/Wallet/controls/CollectibleDetailsHeader.qml:33]: i remember this being discussed as a no no.

**Review by Khushboo-dev-cpp — COMMENTED** [review]
Looks good, I am only have doubts about handling the screen size bby reducing text and icon size as this should ideally come from designs and not form us.

Maybe we need to discuss this with design and John?

File: ui/app/AppLayouts/Wallet/views/RightTabView.qml (hunk 1 above)
Comment (dlipicar) [reply ui/app/AppLayouts/Wallet/views/RightTabView.qml:32]: Oh that's true! I missed that on the design, I'll check with them

File: ui/app/AppLayouts/Wallet/controls/CollectibleDetailsHeader.qml (hunk 2 above)
Comment (dlipicar) [reply ui/app/AppLayouts/Wallet/controls/CollectibleDetailsHeader.qml:23]: Hmm interesting, this was my interpretation of the design. Do you mean what's on [Figma](https://www.figma.com/file/FkFClTCYKf83RJWoifWgoX/Wallet-v2?node-id=6455%3A122769&t=IKRFhXdnDonNMuuh-0) could be outdated?

Comment (Khushboo-dev-cpp) [reply ui/app/AppLayouts/Wallet/controls/CollectibleDetailsHeader.qml:23]: Hey I just saw the design and I didnt know designs for smaller screen size were already added! this is perfect then :)

File: ui/app/AppLayouts/Wallet/views/RightTabView.qml (hunk 1 above)
Comment (dlipicar) [reply ui/app/AppLayouts/Wallet/views/RightTabView.qml:32]: John confirmed it should be "Collectbiles"

File: ui/app/AppLayouts/Wallet/controls/CollectibleDetailsHeader.qml [hunk 4]
```diff
@@ -8,41 +8,87 @@ import StatusQ.Core.Theme 0.1
...
+                value: root.secondaryText
+            }
+        }
+
+        InformationTag {
+            id: networkTag
+            readonly property bool isNetworkValid: networkName !== ""
+            image.source: isNetworkValid && networkIconURL !== "" ? Style.svg("tiny/" + networkIconURL) : ""
+            tagPrimaryLabel.text: isNetworkValid ? networkName : "---"
+            tagPrimaryLabel.color: isNetworkValid ? networkColor : "black"
```
Comment (caybro) [inline ui/app/AppLayouts/Wallet/controls/CollectibleDetailsHeader.qml:90]: How does "black" work in the dark mode, could you post a screenshot? :)

**Review by caybro — COMMENTED** [review]
Got one small doubt :)

File: ui/app/AppLayouts/Wallet/controls/CollectibleDetailsHeader.qml (hunk 4 above)
Comment (caybro) [reply ui/app/AppLayouts/Wallet/controls/CollectibleDetailsHeader.qml:90]: Ah I see, good!

**Review by alaibe — APPROVED** [review]

**Comment by anastasiyaig** [comment]
i've tried to test it but i was not able to make collectibles tab working in wallet, even i can access the OpenSea in browser (both my desktop Chrome and built-in status browser). I am using VPN to access the OpenSea. When i do that, i am seeing the following errors in console:

```
ERR 2022-12-01 10:24:08.160+03:00 error:                                     tid=2627908 file=async_tasks.nim:14 errDesription="\nstatus-go error [methodName:wallet_getOpenseaCollectionsByOwner, code:-32000, message:Get \"https://api.opensea.io/api/v1/collections?asset_owner=0x92D883ed38dEb10DF7C525bB2A7F04d7A00F5C5a&offset=0&limit=300\": context deadline exceeded (Client.Timeout exceeded while awaiting headers) ]\n"
```

```
ERR 2022-12-01 10:16:27.717+03:00 rpc response error                         topics="rpc" tid=2619578 file=core.nim:29 err="\nstatus-go error [methodName:wallet_getOpenseaCollectionsByOwner, code:-32000, message:Get \"https://api.opensea.io/api/v1/collections?asset_owner=0x92D883ed38dEb10DF7C525bB2A7F04d7A00F5C5a&offset=0&limit=300\": read tcp 172.21.0.1:64782->104.18.17.184:443: read: connection reset by peer ]\n"
```

Screenshot 2022-12-01 at 10 25 25


Screenshot 2022-12-01 at 10 23 07

@alaibe @dlipicar sadly this is not really working to me, but assuming the OpenSea is accessible in browser, i suspect we can fix smth on go side to make in work in wallet, please take a look

**Comment by anastasiyaig** [comment]
@glitchminer please take over because i cant test it because of the issue above ( will add is separately)

**Comment by glitchminer** [comment]
Hi @dlipicar , I'm not seeing the collectibles tab being populated. The following error appears to be related to status-go and opensea - please can you take a look?

```
Unhandled wallet signal: recent-history-fetching
ERR 2022-12-06 21:00:46.163+00:00 rpc response error                         topics="rpc" tid=4250323 file=core.nim:29 err="\nstatus-go error [methodName:wallet_getOpenseaCollectionsByOwner, code:-32000, message:invalid character \'E\' looking for beginning of value ]\n"
ERR 2022-12-06 21:00:46.164+00:00 error doing rpc request                    topics="rpc" tid=4250323 file=core.nim:33 methodName=wallet_getOpenseaCollectionsByOwner exception="\nstatus-go error [methodName:wallet_getOpenseaCollectionsByOwner, code:-32000, message:invalid character \'E\' looking for beginning of value ]\n"
ERR 2022-12-06 21:00:46.164+00:00 error:                                     tid=4250323 file=async_tasks.nim:14 errDesription="\nstatus-go error [methodName:wallet_getOpenseaCollectionsByOwner, code:-32000, message:invalid character \'E\' looking for beginning of value ]\n"
Unhandled wallet signal: recent-history-fetching
```

**Comment by anastasiyaig** [comment]
this is the same what i was seeing, which prevented me from testing this change, addressed in  https://github.com/status-im/status-desktop/issues/8545 @glitchminer

**Comment by glitchminer** [comment]
I thought you had some connection issue as well though? Not sure I saw that during my attempt.

**Comment by anastasiyaig** [comment]
I blamed connectivity (actually an access to OpenSea from my location) but seems it is not the case, especially if u can reproduce the same error. The error also is reproducible in master

**Comment by dlipicar** [comment]
> I blamed connectivity (actually an access to OpenSea from my location) but seems it is not the case, especially if u can reproduce the same error. The error also is reproducible in master

ugh.... I'll take a look at this today.
@glitchminer just to be sure, could you try adding these addresses as watch-only?

0x5aB130251D128bc10Df5F77573A8983307822d36
0x648550Eaca1654A1e65B13cf385Ef9Ff505dE909
0xb9eB79E3E735Ee636255dD8D65872a1287744e33
0xBCa12D41FE434fDd4282FF268f9AD67FE44CF535

**Comment by glitchminer** [comment]
@dlipicar , sorry, same issue with all of those addresses. I can see the balance of them but nothing appears under collectibles.

**Comment by dlipicar** [comment]
> @dlipicar , sorry, same issue with all of those addresses. I can see the balance of them but nothing appears under collectibles.

@glitchminer please see https://github.com/status-im/status-desktop/issues/8545#issuecomment-1351751747
Could you try again with a local build?

**Comment by anastasiyaig** [comment]
@dlipicar thanks for being that patient, i was able to cheat infura and finally got to this screen with collectible details :D Few questions though:

1. for network indicator, should we use **Ethereum** (like designs) or **Ethereum Mainnet**?
2. properties names are all lower case, can we please fix that?
3. in some descriptions of tokens there is a markup, for example for this token , Wink word should be italic https://opensea.io/assets/ethereum/0x06012c8cf97bead5deae237070f9587f8e7a266d/1680101
4. is there a task to implement Activity tab? 

Screenshot 2022-12-15 at 11 22 18

**Comment by dlipicar** [comment]
> @dlipicar thanks for being that patient, i was able to cheat infura and finally got to this screen with collectible details :D Few questions though:
> 
> 1. for network indicator, should we use **Ethereum** (like designs) or **Ethereum Mainnet**?
> 2. properties names are all lower case, can we please fix that?
> 3. in some descriptions of tokens there is a markup, for example for this token , Wink word should be italic https://opensea.io/assets/ethereum/0x06012c8cf97bead5deae237070f9587f8e7a266d/1680101
> 4. is there a task to implement Activity tab?
> 
> Screenshot 2022-12-15 at 11 22 18

Thanks for testing!

1) Hmm the design shows "Ethereum", but "Ethereum Mainnet" is how we call the network in the rest of the app and I'm getting the network name from the same place
image
Perhaps it's something we want changed in other places as well? I'll ask Design.

2) Yeah, that was a decision. Before we made them all uppercase, now we're using it exactly as they're provided by the OpenSea API (in this particular case it's all lowercase, but for some others it will have the first letter of each word with uppercase, which is what the design shows).

3) Nice catch! I hope we already have that markup thing implemented somewhere

4) There is since 10 seconds ago: https://github.com/status-im/status-desktop/issues/8811

**Comment by anastasiyaig** [comment]
@dlipicar 

> I hope we already have that markup thing implemented somewhere

I doubt we have it somewhere, but if possible, can u remove it to receive plain text? If it is too much effort then give up :D

**Review by anastasiyaig — APPROVED** [review]

**Comment by dlipicar** [comment]
> @dlipicar
> 
> > I hope we already have that markup thing implemented somewhere
> 
> I doubt we have it somewhere, but if possible, can u remove it to receive plain text? If it is too much effort then give up :D

That's already fixed! We're showing the description in markdown format now. It matches what OpenSea shows on their website.

Screenshot 2022-12-15 at 11 09 53

**Pull request merged**